nautobot/dcim/views.py

```diff
@@ -281,24 +283,23 @@ def get_extra_context(self, request, instance):
             instance.descendants(include_self=True).restrict(request.user, "view").values_list("pk", flat=True)
         )
         stats = {
-            "rack_count": Rack.objects.restrict(request.user, "view").filter(location__in=related_locations).count(),
-            "device_count": Device.objects.restrict(request.user, "view")
-            .filter(location__in=related_locations)
-            .count(),
-            "prefix_count": Prefix.objects.restrict(request.user, "view")
-            .filter(locations__in=related_locations)
-            .count(),
-            "vlan_count": VLAN.objects.restrict(request.user, "view")
+            Rack: Rack.objects.restrict(request.user, "view").filter(location__in=related_locations).count(),
+            Device: Device.objects.restrict(request.user, "view").filter(location__in=related_locations).count(),
+            Prefix: Prefix.objects.restrict(request.user, "view").filter(locations__in=related_locations).count(),
+            VLAN: VLAN.objects.restrict(request.user, "view")
             .filter(locations__in=related_locations)
             .distinct()
             .count(),
-            "circuit_count": Circuit.objects.restrict(request.user, "view")
+            Circuit: Circuit.objects.restrict(request.user, "view")
             .filter(circuit_terminations__location__in=related_locations)
             .count(),
-            "vm_count": VirtualMachine.objects.restrict(request.user, "view")
+            VirtualMachine: VirtualMachine.objects.restrict(request.user, "view")
             .filter(cluster__location__in=related_locations)
             .count(),
         }
+        stats_panel = StatsPanel(
+            label="Stats", section=SectionChoices.RIGHT_HALF, weight=100, filter_name="location", stats=stats
+        )
```

Comment:
Let's see if we can come up with a pattern for including the panel definition in the view's `object_detail_content` alongside the view's other panels, rather than generating it on-the-fly in `get_extra_context()` and passing it as a context variable.